Source/Scene/ParticleSystem.js

```diff
@@ -0,0 +1,539 @@
+     * @param {Number} [options.maximumWidth=1.0] Sets the maximum width of particles in pixels.
+     * @param {Number} [options.height] Sets the minimum and maximum height of particles in pixels.
+     * @param {Number} [options.minimumHeight=1.0] Sets the minimum height of particles in pixels.
+     * @param {Number} [options.maximumHeight=1.0] Sets the maximum height of particles in pixels.
+     * @param {Number} [options.lifeTime=Number.MAX_VALUE] How long the particle system will emit particles, in seconds.
+     *
+     * @demo {@link http://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=ParticleSystem.html|Particle Systems Demo}
+     */
+    function ParticleSystem(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * Whether to display the particle system.
+         * @type {Boolean}
+         * @default true
+         */
+        this.show = defaultValue(options.show, true);
+
+        /**
+         * An array of force callbacks. The callback is passed a {@link Particle} and the difference from the last time
+         * @type {ParticleSystem~applyForce[]}
+         * @default undefined
+         */
+        this.forces = options.forces;
+
+        var emitter = options.emitter;
+        if (!defined(emitter)) {
+            emitter = new CircleEmitter(0.5);
+        }
+        /**
+         * The particle emitter for this system.
+         * @type {ParticleEmitter}
+         * @default CricleEmitter
+         */
+        this.emitter = emitter;
+
+        /**
+         * The 4x4 transformation matrix that transforms the particle system from model to world coordinates.
+         * @type {Matrix4}
+         * @default Matrix4.IDENTITY
+         */
+        this.modelMatrix = Matrix4.clone(defaultValue(options.modelMatrix, Matrix4.IDENTITY));
+        this._modelMatrix = new Matrix4();
+
+        /**
+         * The 4x4 transformation matrix that transforms the particle system emitter within the particle systems local coordinate system.
+         * @type {Matrix4}
+         * @default Matrix4.IDENTITY
+         */
+        this.emitterModelMatrix = Matrix4.clone(defaultValue(options.emitterModelMatrix, Matrix4.IDENTITY));
+        this._emitterModelMatrix = new Matrix4();
+
+        /**
+         * The color of a particle when it is born.
+         * @type {Color}
+         * @default Color.WHITE
+         */
+        this.startColor = Color.clone(defaultValue(options.startColor, Color.WHITE));
+        /**
+         * The color of a particle when it dies.
+         * @type {Color}
+         * @default Color.WHITE
+         */
+        this.endColor = Color.clone(defaultValue(options.endColor, Color.WHITE));
+
+        /**
+         * The scale of the particle when it is born.
+         * @type {Number}
+         * @default 1.0
+         */
+        this.startScale = defaultValue(options.startScale, 1.0);
```

Comment:
What are the bounds on all these?  Do any of these need explicit `set` functions to throw `DeveloperError`s?